e2e/deployment.go

```diff
@@ -296,3 +297,151 @@ func (rnr *rookNFSResource) Do(action kubectlAction) error {
+	}
+
+	return nil
+}
+
+func waitForClusterNameUpdate(c kubernetes.Interface, ns, deploymentName string, timeout int) error {
+	e2elog.Logf("waiting for deployment updates %s/%s", ns, deploymentName)
+
+	// Scale down to 0.
+	scale, err := c.AppsV1().Deployments(ns).GetScale(context.TODO(), deploymentName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("error get scale deployment %s: %w", deploymentName, err)
+	}
+	count := scale.Spec.Replicas
+	scale.ResourceVersion = "" // indicate the scale update should be unconditional
+	scale.Spec.Replicas = 0
+	err = waitForDeploymentUpdateScale(c, ns, deploymentName, scale, timeout)
+	if err != nil {
+		return err
+	}
+
+	// Update deployment.
+	deployment, err := c.AppsV1().Deployments(ns).Get(context.TODO(), deploymentName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("error get deployment %s: %w", deploymentName, err)
+	}
+	containers := deployment.Spec.Template.Spec.Containers
+	for i := range containers {
+		if containers[i].Name == "csi-rbdplugin" || containers[i].Name == "csi-rbdplugin-controller" {
+			containers[i].Args = append(containers[i].Args, fmt.Sprintf("--clustername=%s", defaultClusterName))
+			deployment.Spec.Template.Spec.Containers[i].Args = containers[i].Args
+		}
+	}
+	// clear creationTimestamp, generation, resourceVersion, and uid
+	deployment.CreationTimestamp = metav1.Time{}
+	deployment.Generation = 0
+	deployment.ResourceVersion = "0"
+	deployment.UID = ""
+	err = waitForDeploymentUpdate(c, ns, deploymentName, deployment, timeout)
+	if err != nil {
+		return err
+	}
+
+	// Scale up to count.
+	scale.Spec.Replicas = count
+	err = waitForDeploymentUpdateScale(c, ns, deploymentName, scale, timeout)
+	if err != nil {
+		return err
+	}
+
+	// wait for scale to become count
+	t := time.Duration(timeout) * time.Minute
+	start := time.Now()
+	err = wait.PollImmediate(poll, t, func() (bool, error) {
+		deploy, getErr := c.AppsV1().Deployments(ns).Get(context.TODO(), deploymentName, metav1.GetOptions{})
+		if getErr != nil {
+			if isRetryableAPIError(getErr) {
+				return false, nil
+			}
+			e2elog.Logf(
+				"Deployment Get %s/%s has not completed yet (%d seconds elapsed)",
+				ns, deploymentName, int(time.Since(start).Seconds()))
+
+			return false, fmt.Errorf("error getting deployment %s: %w", deploymentName, getErr)
+		}
+		if deploy.Status.Replicas != count {
+			e2elog.Logf("Expected deplyment %s replicas %d, got %d", deploymentName, count, deploy.Status.Replicas)
+
+			return false, fmt.Errorf("error expected deplyment %s replicas %d, got %d",
+				deploymentName, count, deploy.Status.Replicas)
+		}
+
+		return true, nil
+	})
+	if err != nil {
+		return fmt.Errorf("failed getting deployment %s: %w", deploymentName, err)
+	}
+
+	err = waitForDeploymentComplete(c, deploymentName, ns, timeout)
+	if err != nil {
+		e2elog.Failf("timeout waiting for deployment %s: %v", deploymentName, err)
+	}
```

Comment:
Do we need this check again as `waitForDeploymentUpdateScale` takes care as it makes sure the expected replica are matching?

Reply:
I will remove it, for some reason I thought we need it which I don't seem to recall now.

Reply:
this is still present not removed yet?